pkg/sshd/lock.go

```diff
@@ -0,0 +1,63 @@
+package sshd
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// RepositoryLock interface that allows the creation of a lock associated
+// with a repository name to avoid simultaneous git operations
+type RepositoryLock interface {
+	// Lock aquires a lock for a repository
+	Lock(repoName string, timeout time.Duration) error
+	// Unlock releases the lock for a repository or returns
+	// an error if the specified name doesn't exist.
+	Unlock(repoName string, timeout time.Duration) error
+}
+
+// NewInMemoryRepositoryLock returns a new instance of a key mutex.
```

Comment:
These docs don't seem to be current. What is a "key mutex". Also, it appears that the timeouts aren't respected here - can you add some docs to indicate that fact?